cmd/incusd/images.go

```diff
@@ -1356,18 +1367,20 @@ func doImagesGet(ctx context.Context, tx *db.ClusterTx, recursion bool, projectN
 	var resultMap []*api.Image
 
 	if recursion {
-		resultMap = make([]*api.Image, 0, len(fingerprints))
+		resultMap = make([]*api.Image, 0, len(imagesProjectsMap))
 	} else {
-		resultString = make([]string, 0, len(fingerprints))
+		resultString = make([]string, 0, len(imagesProjectsMap))
 	}
 
-	for _, fingerprint := range fingerprints {
-		image, err := doImageGet(ctx, tx, projectName, fingerprint, public)
+	for fingerprint, projects := range imagesProjectsMap {
+		curProjectName := projects[0]
+		image, err := doImageGet(ctx, tx, curProjectName, fingerprint, public)
 		if err != nil {
 			continue
 		}
 
-		if !image.Public && !hasPermission(auth.ObjectImage(projectName, fingerprint)) {
+		// TODO: Does this work for 'all-projects' since hasPermission i.e. CheckPermission will deny?
```

Comment:
That should be fine, you're providing the project name to the check, so it should do the correct thing and filter the list down to whatever projects the user has read permissions on.

Reply:
Thank you! I also tested in labs, worked fine.